compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -171,7 +171,18 @@ void TR_X86ProcessorInfo::initialize(bool force)
             uint32_t extended_model = getCPUModel(_processorSignature) + (getCPUExtendedModel(_processorSignature) << 4);
             switch (extended_model)
                {
-               case 0x55:
+               case 0xcf:
+                  _processorDescription |= TR_ProcessorIntelEmeraldRapids; break;
+               case 0x8f:
+                  _processorDescription |= TR_ProcessorIntelSapphireRapids; break;
+               case 0x6a:  // IceLake_X
+               case 0x6c:  // IceLake_D
+               case 0x7d:  // IceLake
+               case 0x7e:  // IceLake_L
+                  _processorDescription |= TR_ProcessorIntelIceLake; break;
+               case 0x55:  // Skylake_X
```

Comment:
I'm not sure how skylake ```0x55``` will be differentiated from Cooper lake and Cascade lake. The model numbers overlap.

Reply:
Both are considered "optimizations" on Skylake.  I believe the stepping is what distinguishes them.  I didn't include them because I did not find a verifiable stepping value for both, and distinguishing the processor with that much refinement was not my goal (it may be part of your future CPUID refactoring work).

Nevertheless I have since found a stepping value for Cascade Lake, and I modified the PR to support it.